### ISSUE TITLE
\ref{} is detected as end of a sentence

### ISSUE THREAD

**Issue by atsuyaw** [textlint/textlint-plugin-latex2e#168]
When I use `\ref{}`, the command was detected as end of a sentence:  
![image](https://user-images.githubusercontent.com/68371029/119588090-043daa80-be0b-11eb-99ac-fa380a165ba0.png)

It is expected to detect an error in sentences such as line 199, but errors outed in line 200 and 201.

`\ref{}` command is used for cross-reference frequently in academic writing.  
In fact, such structure of sentences were contained a lot in my file, so the output were filled with these errors.

**Comment by pddg** [comment]
@atsuyaw 
Thank you for reporting the issue. I confirmed that it is a bug.
Sorry for the delay in addressing the other reported issues 😥 .
We will do our best to fix it.

**Comment by pddg** [comment]
## How to reproduce

### Environment

- Node.js v14.17.0
- textlint-plugin-latex2e v1.1.3
- textlint v12.0.0
- textlint-rule-ja-no-mixed-period v2.1.1

### Procedure

```bash
$ mkdir textlint-latex2e-issue168 && cd $_
$ npm init -y
$ npm install --save \
    textlint@12.0.0 \
    textlint-plugin-latex2e@1.1.3 \
    textlint-rule-ja-no-mixed-period@2.1.1
$ cat << EOF > sample.tex
本日は図\ref{fig:sunnyy}に示すように晴天なり。
EOF
$ cat << EOF > .textlintrc
{
    "plugins": ["latex2e"],
    "rules": {
        "ja-no-mixed-period": {
            "periodMark": "。"
        }
    }
}
EOF
$ textlint sample.tex

/Users/pudding/Factory/textlint-latex2e-issue168/sample.tex
  1:4  error  文末が"。"で終わっていません。  ja-no-mixed-period

✖ 1 problem (1 error, 0 warnings)
```

## Cause of this issue

There seems to be an error in the way `Paragraph` is interpreted. There should be an inline node and a block node, but since `\ref` is interpreted as a block node, there are two separate Paragraphs, which should be one.

```
$ npx tex2tast -r sample.tex
{
  "type": "Document",
  "raw": "本日は図\\ref{fig:sunnyy}に示すように晴天なり。\n",
  "children": [
    {
      "raw": "本日は図",
      "type": "Paragraph",
      "children": [
        {
          "raw": "本日は図",
          "type": "Str",
          "value": "本日は図"
        }
      ]
    },
    {
      "raw": "\\ref{fig:sunnyy}",
      "value": "fig:sunnyy",
      "type": "Html"
    },
    {
      "raw": "に示すように晴天なり。",
      "type": "Paragraph",
      "children": [
        {
          "raw": "に示すように晴天なり。",
          "type": "Str",
          "value": "に示すように晴天なり。"
        }
      ]
    }
  ]
}
```

This problem can be avoided by interpreting it as an inline element, just like label and url.

**Comment by pddg** [comment]
hmm...

This problem is caused by #70 . HTML type nodes contain two types, inline and block, but there is no way to tell the difference, so they are always interpreted as block elements.
https://github.com/textlint/textlint-plugin-latex2e/blob/8bd09103dd6a3a4cd6d1588c5398093499e34537/src/paragraphize.ts#L42

@tani Do you know of any good solutions?

**Comment by pddg** [comment]
Oh, I found `HtmlBlock` in ASTNodeTypes definitions.

https://github.com/textlint/textlint/blob/master/packages/%40textlint/ast-node-types/src/index.ts#L26

We need to use both `Html` and `HtmlBlock`.

**Comment by pddg** [comment]
@atsuyaw 
We published a new version of textlint-plugin-latex2e v1.1.4. It includes some fixes for #158 and this issue.
Please check and let me know if there are any bugs. Thank you!

**Comment by atsuyaw** [comment]
@pddg
Thank you for your corresponding!  
I confirmed the problems solved.
